### ISSUE TITLE
CA2000 false positive inside foreach loop

### ISSUE THREAD

**Issue by batkaevruslan** [dotnet/roslyn-analyzers#7047]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0 (Latest)

### Describe the bug

I believe it is a false positive error: analyzers issues an error inside the loop, but does not issue any error when the same code is used outside the loop.

### Steps To Reproduce

```
public class Test
{
    public static async Task ConvertWithLoopAsync()
    {
        foreach (int _ in Enumerable.Range(0, 1))
        {
            using Foo pdfDocument = new(); //CA2000 => object created by 'new()' should be disposed
            await PrepareForRenderingAsync();
        }
    }

    public static async Task ConvertAsync()
    {
        using Foo pdfDocument = new(); //OK
        await PrepareForRenderingAsync();
    }

    private static async Task PrepareForRenderingAsync()
    {
        await using Stream targetStream = File.Create("");
    }

    private sealed class Foo : IDisposable
    {
        public void Dispose()
        {
        }
    }
}
```

### Expected behavior

No CA2000

### Actual behavior

CA2000
![image](https://github.com/dotnet/roslyn-analyzers/assets/3961279/c027b997-701f-4236-98cc-8e0c7f4d996e)

**Comment by Boomkop3** [comment]
If you need a quick workaround: `.ToArray().Span()`. Depending on your situation you might not need `.ToArray()` such as when iterating over a Dictionary<>

**Comment by Boomkop3** [comment]
Converting your collection to an array and using a for loop can be a bandaid here

**Comment by TheConstructor** [comment]
One can probably write an `ForEachAsync`-extension or put the block-contents into a local function. It feels like the loop-body isn't properly detected as scope in an async context
